scripts/behavior.js

```diff
@@ -1470,13 +1470,21 @@ function formatIrcMessage(authorString, messageString){
 }
 
 /// Sends a message received from the IRC
-function sendFromIrc(channelName, authorString, messageString){
-	for (let i = 0; i < receivers[channelName].length; i++){
-		let channel = receivers[channelName][i];
-		if (channel != undefined){
-			sendMessage(channel, '**'+authorString+'**: '+messageString);
-		}
+function sendFromIrc(channelName, authorString, messageString) {
+    if (!receivers[channelName]) {
+        return;
 	}
+    let sendingTimes = 0;
+
+    let usedChannels = []; //
+	for (let i = 0; i < receivers[channelName].length; i++) {
+        let channel = receivers[channelName][i];
+        if (channel !== undefined && usedChannels.indexOf(channelName) === -1) {
+            usedChannels.push(channelName);
+            sendingTimes += 1;
+            sendMessage(channel, '**' + authorString + '**: ' + messageString);
+        }
+    }

```

Comment:
These changes are equivalent to doing the loop just with i = 0. That's certainly not a tight fix, I imagine multiple receivers are there to propagate IRC to different Discord channels? This change would break this.

Reply:
actually it's hopefix :-)
but it won't work. @Askaholic has found the reason, but we don't have solution yet